### PR TITLE
Support new upsample in symbolic, caffe2 backend & caffe2 frontend

### PR DESCRIPTION
**Pull request by zrphercule**
We updated the description of upsample_op in onnx: https://github.com/onnx/onnx/pull/1467
Therefore, we need to support the new upsample_op in caffe2-onnx backend as well.

### CODE REVIEW

File: caffe2/onnx/helper.h [hunk 1]
```diff
@@ -12,6 +12,7 @@ namespace onnx {
 
 using ::ONNX_NAMESPACE::AttributeProto;
 using ::ONNX_NAMESPACE::NodeProto;
+using ::ONNX_NAMESPACE::TensorProto;
```
Comment (houseroad) [inline caffe2/onnx/helper.h]: Using TensorProto here is not recommended, especially, it's a header file.
And another reason is that we have both caffe2::TensorProto and onnx::TensorProto (caffe2 does not have NodeProto and AttributeProto). I would suggest explicitly use onnx::TensorProto in the following code.

File: caffe2/onnx/helper.h [hunk 2]
```diff
@@ -71,6 +72,25 @@ inline AttributeProto MakeAttribute(
   return attr;
 }
 
+inline AttributeProto MakeAttribute(const std::string& name, TensorProto& val) {
+  AttributeProto attr;
+  attr.set_name(name);
+  std::string val_buf;
+  *(attr.mutable_t()) = val;
```
Comment (houseroad) [inline caffe2/onnx/helper.h]: Use CopyFrom instead of direct assign

File: caffe2/onnx/helper.h [hunk 3]
```diff
@@ -71,6 +72,25 @@ inline AttributeProto MakeAttribute(
   return attr;
 }
 
+inline AttributeProto MakeAttribute(const std::string& name, TensorProto& val) {
+  AttributeProto attr;
+  attr.set_name(name);
+  std::string val_buf;
```
Comment (houseroad) [inline caffe2/onnx/helper.h]: unused?

File: caffe2/onnx/helper.h [hunk 4]
```diff
@@ -71,6 +72,25 @@ inline AttributeProto MakeAttribute(
...
+  *(attr.mutable_t()) = val;
+  attr.set_type(AttributeProto::TENSOR);
+  return attr;
+}
+
+inline TensorProto MakeTensor(const string& name, const std::vector<float>& v) {
+  TensorProto ret;
+  ret.set_name(name);
+  ret.add_dims(v.size());
+  ret.set_data_type(TensorProto::FLOAT);
```
Comment (houseroad) [inline caffe2/onnx/helper.h]: Can we make it general? vector template with char, and pass the data_type as parameter?

**Review by houseroad — COMMENTED** [review]
Some inline comments

File: caffe2/onnx/helper.h (hunk 1 above)
Comment (zrphercule) [reply caffe2/onnx/helper.h]: Yeah that's a good idea.

Comment (zrphercule) [reply caffe2/onnx/helper.h]: wait, you mean ONNX_NAMESPACE::TensorProto bot onnx::TensorProto, right?

File: caffe2/onnx/helper.h (hunk 4 above)
Comment (zrphercule) [reply caffe2/onnx/helper.h]: I guess we can, if data_type is passed as parameter.

**Comment by zrphercule** [comment]
@houseroad Any more comments? CI is green.

**Review by houseroad — APPROVED** [review]

**Comment by houseroad** [comment]
import and update the submodule internally?

**Pull request closed without merging**